Userland/Utilities/comm.cpp

```diff
@@ -58,31 +61,38 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
         return 1;
     }
 
-    auto process_file = [](String const& path, auto& file, int file_number) {
-        if (path == "-") {
-            file = Core::File::standard_input();
-        } else {
-            auto file_or_error = Core::File::open(path, Core::OpenMode::ReadOnly);
-            if (file_or_error.is_error()) {
-                warnln("Failed to open file{} '{}': {}", file_number, path, file_or_error.error());
-                return false;
-            }
-            if (Core::File::is_directory(path)) {
-                warnln("Failed to open file{} '{}': is a directory", file_number, path);
-                return false;
-            }
-            file = file_or_error.value();
+    auto open_file = [](String const& path, auto& file, int file_number) {
+        auto file_or_error = Core::Stream::File::open_file_or_standard_stream(path, Core::Stream::OpenMode::Read);
+        if (file_or_error.is_error()) {
+            warnln("Failed to open file{} '{}': {}", file_number, path, file_or_error.error());
+            return false;
+        }
+
+        if (Core::File::is_directory(path)) {
+            warnln("Failed to open file{} '{}': is a directory", file_number, path);
+            return false;
+        }
```

Comment:
This might slightly change the behavior if you have a directory named `-`.
Since that gets STDIN but then here we do check if it's a directory where we only did that check for `path != "-"` before